src/utils/apiRequests/api.ts

```diff
@@ -249,25 +277,33 @@ export async function postAuthenticatedRequest<T>(
 export async function deleteAuthenticatedRequest(
   url: any,
   token: any,
-  impersonatedEmail?: string,
   errorHandler?: Function
 ): Promise<any> {
+  const targetData = localStorage.getItem("impersonationData") ;
+  const  impersonatedDataFromLocal = JSON.parse(targetData);
```

Comment:
You can probably combine these 2 lines. Don't forget to check for TS warnings - JSON.parse expects a string.